pumps/kafka.go

```diff
@@ -61,11 +70,33 @@ func (k *KafkaPump) Init(config interface{}) error {
 		}
 	}
 
+	var mechanism sasl.Mechanism
+
+	switch k.kafkaConf.SASLMechanism {
+	case "":
+		break
+	case "PLAIN", "plain":
+		mechanism = plain.Mechanism{Username: k.kafkaConf.Username, Password: k.kafkaConf.Password}
+	case "SCRAM", "scram":
+		algorithm := scram.SHA256
+		if k.kafkaConf.Algorithm == "sha-512" || k.kafkaConf.Algorithm == "SHA-512" {
+			algorithm = scram.SHA512
+		}
+		var mechErr error
+		mechanism, mechErr = scram.Mechanism(algorithm, k.kafkaConf.Username, k.kafkaConf.Password)
+		if mechErr != nil {
+			k.log.Fatal("Failed initialize kafka mechanism  : ", mechErr)
+		}
+	default:
+		k.log.Warn("Tyk pump doesn't support " + k.kafkaConf.SASLMechanism + " mechanism.")
+	}
+
```

Comment:
Can we add a warning message if we have mechanism set AND tlsConfig not set?

Reply:
sure!